### DISCUSSION TITLE
How to get room events with dynamic namespaces

### DISCUSSION THREAD

**Discussion by hiapk** [socketio/socket.io#4757] [Q&A] [answered]
io.of("/").adapter.on("create-room", (room) => {
  console.log(`room ${room} was created`);
});

hello developers, how to get  "create-room" event  when i use  dynamic namespaces ?  
thanks

**Comment by darrachequesne** [comment, marked as answer]
Hi! You can listen to the `new_namespace` event:

```js
io.on("new_namespace", (namespace) => {
  namespace.adapter.on("create-room", (room) => {
    console.log(room ${room} was created);
  });
});
```

Reference: https://socket.io/docs/v4/server-api/#event-new_namespace
